src/main/scala/com/yotpo/metorikku/instrumentation/KafkaLagWriter.scala

```diff
@@ -0,0 +1,41 @@
+package com.yotpo.metorikku.instrumentation
+
+import java.util
+
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import org.apache.kafka.clients.consumer.{KafkaConsumer, OffsetAndMetadata}
+import org.apache.kafka.common.TopicPartition
+import org.apache.spark.sql.streaming.StreamingQueryListener
+import org.apache.spark.sql.streaming.StreamingQueryListener._
+
+class KafkaLagWriter(kafkaConsumer: KafkaConsumer[String, String], topic: String) extends StreamingQueryListener {
```

Comment:
Have you tested this on a cluster?

Reply:
?

Reply:
@lyogev tested on a cluster and its working :)